backend/src/v2/compiler/argocompiler/container.go

```diff
@@ -121,6 +122,24 @@ func (c *workflowCompiler) addContainerDriverTemplate() string {
 			Resources: driverResources,
 		},
 	}
+	// Add retry policy
+	if retryPolicy != nil {
+		maxRetryCount := intstr.FromInt(int(retryPolicy.GetMaxRetryCount()))
+		// During the converstion from float64 to int, the fraction is discarded (truncated towards 0).
+		// See https://go.dev/ref/spec#Conversions
+		backOffFactor := intstr.FromInt(int(retryPolicy.GetBackoffFactor()))
+		if maxRetryCount.IntValue() > 0 {
+			// If maxRetryCount == 0, that means retry policy is empty. In this case we leave the whole field empty.
+			t.RetryStrategy = &wfapi.RetryStrategy{
+				Limit: &maxRetryCount,
+				Backoff: &wfapi.Backoff{
+					Duration:    retryPolicy.GetBackoffDuration().String(),
+					Factor:      &backOffFactor,
+					MaxDuration: retryPolicy.GetBackoffMaxDuration().String(),
```

Comment:
We need to check the value of `retryPolicy.GetBackoffMaxDuration()` and replace it with `1h` or `3600s` if it is nil or set to `0`. Ref: https://github.com/kubeflow/pipelines/blob/master/api/v2alpha1/pipeline_spec.proto#L546-L549